synced_collections/numpy_utils.py

```diff
@@ -84,11 +84,11 @@ def _is_numpy_scalar(data):
 def _is_complex(data):
     """Check if an object is complex.
 
-    This function works for both numpy raw Python data types.
+    This function works for numpy arrays, scalars, and raw Python data types.
 
     Returns
     -------
     bool
         Whether or not the input is a complex number.
     """
-    return (NUMPY and numpy.iscomplex(data).any()) or (isinstance(data, complex))
+    return (NUMPY and numpy.iscomplexobj(data)) or isinstance(data, complex)
```

Comment:
This was a bug in the previous implementation where complex scalars weren't handled properly.

```python
>>> import numpy
>>> value = numpy.complex64(0.5+0j)
>>> numpy.iscomplex(value).any()
False
>>> isinstance(value, complex)
False
>>> numpy.iscomplexobj(value)
True
```